File: utils/onebot/help.py
from nonebot.log import logger
from utils.rendering.help import render_help, help_to_text
from utils.rendering.models import RenderError
from .media import image_segment


async def send_help(matcher, document, *, theme="auto", force=False):
    try:
        rendered = await render_help(document, theme=theme, force=force)
    except RenderError:
        logger.exception("Help image unavailable; sending the same document as text")
        text = help_to_text(document)
        for start in range(0, len(text), 2000):
            await matcher.send(text[start : start + 2000])
        return
    for page in rendered.pages:
        await matcher.send(image_segment(page.data))

File: utils/onebot/help.go
package onebot

import (
	"context"
	"fmt"

	"github.com/HakuchumuHYX/HakuBot/utils"
	"github.com/HakuchumuHYX/HakuBot/utils/rendering"
	zero "github.com/wdvxdr1123/ZeroBot"
	"github.com/wdvxdr1123/ZeroBot/message"
)

func SendHelp(ctx context.Context, bot *zero.Ctx, browser *rendering.Browser, paths utils.Paths, cache *rendering.Cache, doc rendering.HelpDocument, theme string) error {
	rendered, err := browser.Help(ctx, paths, cache, doc, theme, false)
	if err != nil {
		if bot.Send(rendering.HelpText(doc)).ID() == 0 {
			return fmt.Errorf("help render failed (%w) and text delivery failed", err)
		}
		return nil
	}
	for _, p := range rendered.Pages {
		if bot.SendChain(message.ImageBytes(p.Data)).ID() == 0 {
			return fmt.Errorf("help image delivery failed")
		}
	}
	return nil
}
